### PR TITLE
GitHub Actions workflow for Fonda

### PR DESCRIPTION
**Pull request by madmongoose**
Stored in .github/workflows/main.yml;
Start triggers: push to develop branch or manually start in Actions page;
500mb storage;
2000 min per month for jobs;
Total concurrent Linux jobs: 20;
Maximum concurrent macOS jobs: 5;
Artefact storage on a jobs web page: from 0 to 90 days;

Supported runners and hardware resources
Hardware specification for Windows and Linux virtual machines:
* 2-core CPU
* 7 GB of RAM memory
* 14 GB of SSD disk space
Hardware specification for macOS virtual machines:
* 3-core CPU
* 14 GB of RAM memory
* 14 GB of SSD disk space

Action steps:
1. Runing docker with ubuntu-latest;
2. Git clone repository;
3. Setup Java JDK 8 (temirin prebuild);
4. Building with Gradle (./gradlew -PbuildNumber=ga.${{ github.run_number }}.${{ github.sha }} build zip --no-daemon);
5. Testing with Gradle (./gradlew jacocoTestReport)
6. Uploading artefact to job web page in zip;
7. Geting AWS credentials from GitHub Secrets;
8. Copying an artefact to AWS S3 Bucket to folder like a branch with name fonda-2.0.0.ga.${{ github.run_number }}.${{ github.sha }}.zip (for example fonda-2.0.0.ga.468b49764403fe39c964a470d270389341dc3014.zip);
9. Automatic clean tasks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![Build status](https://ci.appveyor.com/api/projects/status/4m3y9tyb08n4xllw?svg=true)](https://ci.appveyor.com/project/epam/fonda)
+[![Build status](https://github.com/madmongoose/fonda/workflows/build/badge.svg?branch=develop)](https://github.com/madmongoose/fonda/actions)
```
Comment (sidoruka) [inline README.md]: Please change to EPAM repository, instead of the fork URL

**Review by sidoruka — CHANGES_REQUESTED** [review]
Please also update add the Github Action releases URL to the README (where the user can grab the distro besides the S3)

**Comment by madmongoose** [comment]
Added release action to workflow;
Added releases page to the README;
All commits were squashed.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[![Build status](https://ci.appveyor.com/api/projects/status/4m3y9tyb08n4xllw?svg=true)](https://ci.appveyor.com/project/epam/fonda)
+[![Build status](https://github.com/epam/fonda/workflows/build/badge.svg?branch=develop)](https://github.com/madmongoose/fonda/actions)
```
Comment (kamyshova) [inline README.md]: Change URL to EPAM repository, please

File: .github/workflows/fonda-ci.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Java JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: "temurin" # See 'Supported distributions' for available options
+          java-version: "8"
+          java-package: jdk # optional (jdk or jre) - defaults to jdk
+          cache: "gradle"
```
Comment (kamyshova) [inline .github/workflows/fonda-ci.yml]: Should we use cache?

File: .github/workflows/fonda-ci.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+#---------------------------------------------------------------------
+
+name: build
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the develop branch
+  push:
+    branches: [ develop ]
+    paths-ignore:
+      - "docs/**"
```
Comment (kamyshova) [inline .github/workflows/fonda-ci.yml]: Do you mean `doc` instead of `docs`?

**Review by kamyshova — CHANGES_REQUESTED** [review]

**Review by kamyshova — APPROVED** [review]

**Pull request merged**